### PR TITLE
AST-3616 - Embed(Twitter, soundcloud, etc) block not working on page.

### PR DESCRIPTION
**Pull request by Sudhanshu-1508**
### Description

[Task](https://brainstormforce.atlassian.net/browse/AST-3616)

### Types of changes

Non-breaking changes
### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/compatibility/class-astra-gutenberg.php [hunk 1]
```diff
@@ -159,14 +159,14 @@ public function group_block_replace_regex( $matches ) {
...
-		if ( isset( $block['blockName'] ) && 'core/embed' !== $block['blockName'] && 'core/youtube' !== $block['blockName'] ) {
-			return $block_content;
-		}
-
 		/** @psalm-suppress PossiblyUndefinedStringArrayOffset */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
 		if ( ( ! empty( $block['blockName'] ) && ( 'core/embed' === $block['blockName'] || 'core/youtube' === $block['blockName'] ) ) && ! empty( $block['attrs'] ) && empty( $block['attrs']['url'] ) ) {
 			return $block_content;
 		}
+		
+		if ( ! empty( $block['blockName'] ) && ( 'core/embed' === $block['blockName'] )  && ! empty( $block['attrs']['url'] ) ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-gutenberg.php:167]:
```suggestion
		if ( ! empty( $block['blockName'] ) && ( 'core/embed' === $block['blockName'] ) && ! empty( $block['attrs']['url'] ) ) {
```

File: inc/compatibility/class-astra-gutenberg.php [hunk 2]
```diff
@@ -159,14 +159,14 @@ public function group_block_replace_regex( $matches ) {
...
-		if ( isset( $block['blockName'] ) && 'core/embed' !== $block['blockName'] && 'core/youtube' !== $block['blockName'] ) {
-			return $block_content;
-		}
-
 		/** @psalm-suppress PossiblyUndefinedStringArrayOffset */ // phpcs:ignore Generic.Commenting.DocComment.MissingShort
 		if ( ( ! empty( $block['blockName'] ) && ( 'core/embed' === $block['blockName'] || 'core/youtube' === $block['blockName'] ) ) && ! empty( $block['attrs'] ) && empty( $block['attrs']['url'] ) ) {
 			return $block_content;
 		}
+		
+		if ( ! empty( $block['blockName'] ) && ( 'core/embed' === $block['blockName'] ) && ! empty( $block['attrs']['url'] ) ) {
```
Comment (imnavanath) [inline inc/compatibility/class-astra-gutenberg.php:167]: This condition `! empty( $block['attrs']['url'] )` should be like this right `empty( $block['attrs']['url'] )` ?

Comment (imnavanath) [reply inc/compatibility/class-astra-gutenberg.php:167]: @Sudhanshu-1508 ^

Comment (Sudhanshu-1508) [reply inc/compatibility/class-astra-gutenberg.php:167]: @imnavanath the suggested code triggers and renders only yoututbe block on the frontend, but the newly modified condition makes sure that every blcok is rendered on the frontend.

File: changelog.txt [hunk 3]
```diff
@@ -10,6 +13,7 @@ v4.5.0
 - Fix: Footer Builder - Above and Below Footer vertical alignment options are not working.
 - Fix: EDD - Mini cart widget UI broken in the RTL mode.
 - Fix: Enable Smooth Scroll to ID causes a jerk issue on the comment navigation section.
+- Fix: Embed block (Twitter, SoundCloud, Spotify) does not appear on the frontend.
```
Comment (imnavanath) [inline changelog.txt:16]:
```suggestion
```

File: changelog.txt [hunk 4]
```diff
@@ -1,5 +1,8 @@
 v4.5.1 (Unreleased)
 - Improvement: Compatibility with WooCommerce 8.3.
+- Improvement: Introducing a new filter 'astra_embed_wrapper_class' to add a class to the iframe wrapper for embed blocks.
+- Fix: Function parameter value was wrong in 'responsive_oembed_wrapper' callback while processing embed blocks iframe. (Props: https://github.com/IngoWinter)
+- Fix: SoundCloud, Spotify embed blocks not working on the frontend.
```
Comment (imnavanath) [inline changelog.txt:5]:
```suggestion
- Fix: Embed block (Twitter, SoundCloud, Spotify) does not appear on the frontend.
```

File: inc/compatibility/class-astra-gutenberg.php [hunk 5]
```diff
@@ -24,7 +24,7 @@ public function __construct() {
 		}
 
 		add_filter( 'render_block_core/group', array( $this, 'add_inherit_width_group_class' ), 10, 2 );
-		add_filter( 'render_block', array( $this, 'add_iframe_wrapper' ), 10, 2 );
+		add_filter( 'render_block', array( $this, 'add_iframe_wrapper' ), 99, 2 );
```
Comment (imnavanath) [inline inc/compatibility/class-astra-gutenberg.php:27]:
```suggestion
		add_filter( 'render_block', array( $this, 'add_iframe_wrapper' ), 10, 2 );
```

**Review by imnavanath — APPROVED** [review]

File: changelog.txt (hunk 3 above)
Comment (imnavanath) [inline changelog.txt:16]:
```suggestion
```

File: changelog.txt (hunk 4 above)
Comment (imnavanath) [inline changelog.txt:5]:
```suggestion
- Fix: Embed block (Twitter, SoundCloud, Spotify) does not appear on the frontend.
```

File: inc/compatibility/class-astra-gutenberg.php (hunk 5 above)
Comment (imnavanath) [inline inc/compatibility/class-astra-gutenberg.php:27]:
```suggestion
		add_filter( 'render_block', array( $this, 'add_iframe_wrapper' ), 10, 2 );
```

**Pull request merged**